packages/arb-token-bridge-ui/src/util/TokenWithdrawalUtils.ts

```diff
@@ -24,9 +32,38 @@ export async function withdrawTokenEstimateGas({
     from: address
   })
 
-  const estimatedL2Gas = await l2Provider.estimateGas(
-    withdrawalRequest.txRequest
+  // if it falls into the `0x` case, user shouldn't be able to perform the tx
+  // so it doesn't matter if the estimate is lower than the actual gas
+  const withdrawalRequestCalldata = isHex(withdrawalRequest.txRequest.data)
+    ? withdrawalRequest.txRequest.data
+    : '0x'
+
+  const gasComponents = await arbPublicClient.simulateContract({
+    address: NODE_INTERFACE_ADDRESS,
+    abi: nodeInterfaceABI,
+    functionName: 'gasEstimateL1Component',
+    args: [ARB_RETRYABLE_TX_ADDRESS, false, withdrawalRequestCalldata],
+    account: address
+  })
+
+  // This is the gas needed to execute the withdrawal on L1; not the batch posting fee
+  // 1st result in the array is `gasEstimateForL1` according to ABI
+  const estimatedL1Gas = BigNumber.from(gasComponents.result[0])
+
+  const estimatedTotalGas = BigNumber.from(
```

Comment:
We can get both L1 and L2 gas components via a single contract call instead of deriving it separately. Calling `gasEstimateComponents` (instead of `gasEstimateL1Component`) method gives us `gasEstimate` and `gasEstimateForL1`. 

The L2 gas estimate is `const gasEstimateForL2: BigNumber = gasComponents.gasEstimate.sub(
      gasComponents.gasEstimateForL1
    )`

(Note: pls check `private async estimateArbitrumGas` method in SDK.)

Reply:
good point
I was using a different destination address so the simulated call for `gasEstimate` would fail
Changed it to match the withdrawal request and updated the calculation now